app/code/Ludmila/LSNewClasses/Block/Homework11/Homework_OOP.php

```diff
@@ -0,0 +1,73 @@
+<?php
+namespace Ludmila\LSNewClasses\Block\Homework11;
+use Ludmila\LSNewClasses\Model\ShowFiles;
+use Ludmila\LSNewClasses\Model\ShowConstants;
+use Ludmila\LSNewClasses\Model\ShowTypes;
+
+/**
+ * Class HomeworkOOP
+ * @package Ludmila\LSNewClasses\Block\Homework11
+ */
+class HomeworkOOP extends \Magento\Framework\View\Element\Template {
+    /**
+     * @var ShowFiles
+     */
+    public $showFilesGet;
+    /**
+     * @var ShowConstants
+     */
+    public $showConstantsGet;
+    /**
+     * @var ShowTypes
+     */
+    public $showTypesGet;
+
+    /**
+     * HomeworkOOP constructor.
+     * @param \Magento\Framework\View\Element\Template\Context $context
+     * @param ShowConstants $showConstants
+     * @param ShowFiles $showFiles
+     * @param ShowTypes $showTypes
+     */
+    public function __construct(
+        \Magento\Framework\View\Element\Template\Context $context,
+        ShowConstants $showConstants,
+        ShowFiles $showFiles,
+        ShowTypes $showTypes
+    ) {
+        parent::__construct($context);
+        $this->showFilesGet = $showFiles;
+        $this->showConstantsGet = $showConstants;
+        $this->showTypesGet = $showTypes;
+    }
+    /**
+     * @return \RecursiveIteratorIterator
+     */
+    public function giveFileList(): \RecursiveIteratorIterator
+    {
+        return $this->showFilesGet->giveFileList();
```

Comment:
There is an error in this line. The class Ludmila\LSNewClasses\Model\ShowFiles doesn't have the method giveFileList. THe class  Ludmila\LSNewClasses\Model\ShowFiles has only one method: show()
Did you wanted to call it? like this:
return $this->showFilesGet->show();